### PR TITLE
Create block index views in `FixturesConsole`

### PR DESCRIPTION
**Pull request by thomasdax98**
Corresponds to https://github.com/vivid-planet/comet/pull/1966

The block index views must be (re)created 

- after executing database migrations
- after executing the fixtures (because they drop the whole database and recreate it)

### CODE REVIEW

File: api/package.json [hunk 1]
```diff
@@ -10,7 +10,9 @@
         "db:migrate": "npm run console migrate --",
         "db:migrate:prod": "npm run console:prod migrate --",
         "fixtures": "npm run console fixtures --",
+        "postfixtures": "npm run console createBlockIndexViews",
         "fixtures:prod": "npm run console:prod fixtures --",
+        "postfixtures:prod": "npm run console createBlockIndexViews",
```
Comment (thomasdax98) [inline api/package.json:15]: I'm not sure when the prod fixtures are executed and if we need that here?

**Comment by nsams** [comment]
please explain the problem you want to solve and where it happens

File: api/package.json (hunk 1 above)
Comment (johnnyomair) [reply api/package.json:15]: Maybe they are executed in non-prod environments in some projects?

**Comment by thomasdax98** [comment]
TODO: Call create view in fixtures console

File: api/src/db/migrate.console.ts [hunk 2]
```diff
@@ -70,6 +71,7 @@ export class MigrateConsole {
             this.logger.log(`Executed migrations. Trying to commit...`);
             await em.commit();
             this.logger.log("Migrations successfully committed");
+            await this.dependenciesService.createViews();
```
Comment (nsams) [inline api/src/db/migrate.console.ts:74]: in production, the createViews is done in a separate Job

- if you change that here, that job must be removed
- if i remember correctly, the reason for the job is that it doesn't block api startup (and can run in background) -  migrations must block

File: api/src/db/fixtures/fixtures.console.ts [hunk 3]
```diff
@@ -40,6 +41,8 @@ export class FixturesConsole {
         // Add your fixtures here
         multiBar.stop();
 
+        await this.dependenciesService.createViews();
```
Comment (nsams) [inline api/src/db/fixtures/fixtures.console.ts:44]: :+1:

File: api/src/db/migrate.console.ts [hunk 4]
```diff
@@ -70,6 +71,10 @@ export class MigrateConsole {
             this.logger.log(`Executed migrations. Trying to commit...`);
             await em.commit();
             this.logger.log("Migrations successfully committed");
+
+            if (process.env.NODE_ENV === "development") {
+                await this.dependenciesService.createViews();
```
Comment (johnnyomair) [inline api/src/db/migrate.console.ts:76]: > after executing database migrations

Why? Because of possible new entities with root blocks?

Comment (thomasdax98) [reply api/src/db/migrate.console.ts:76]: Yes

**Review by johnnyomair — DISMISSED** [review]

File: api/src/db/migrate.console.ts (hunk 2 above)
Comment (nsams) [reply api/src/db/migrate.console.ts]: It is not very obvious that migrate command does under a NODE_ENV condition a createViews. I'd keep that like it was before clearly separated and handled in package.json.

**Review by dkarnutsch — APPROVED** [review]

**Review by johnnyomair — APPROVED** [review]

**Review by dkarnutsch — APPROVED** [review]

File: api/src/db/migrate.console.ts (hunk 2 above)
Comment (thomasdax98) [reply api/src/db/migrate.console.ts]: Done in bab79e59fae4c0f83b472ad16ed86b3b20da4fc0

**Review by nsams — APPROVED** [review]

**Pull request merged**